Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m

```diff
@@ -173,4 +176,52 @@ - (NSString *)getActionName {
     return @"open";
 }
 
+
+#pragma - Open Response Lock Handling
+
+
+//	Instead of semaphores, the lock is handled by scheduled dispatch_queues.
+//	This is the 'new' way to lock and is handled better optimized for iOS.
+//	Also, since implied lock is handled by a scheduler and not a hard semaphore it's less error
+//	prone.
+
+
+static dispatch_queue_t openRequestWaitQueue = NULL;
+static BOOL openRequestWaitQueueIsSuspended = NO;
+
+
++ (void) initialize {
+    if (self != [BranchOpenRequest self])
+        return;
+    openRequestWaitQueue =
+        dispatch_queue_create("io.branch.sdk.openqueue", DISPATCH_QUEUE_CONCURRENT);
+}
+
++ (void) setWaitNeededForOpenResponseLock {
+    @synchronized (self) {
+        if (!openRequestWaitQueueIsSuspended) {
+            NSLog(@"Suspend openRequestWaitQueue.");
+            openRequestWaitQueueIsSuspended = YES;
+            dispatch_suspend(openRequestWaitQueue);
+        }
+    }
+}
+
++ (void) waitForOpenResponseLock {
+    NSLog(@"Wait for openRequestWaitQueue.");
+    dispatch_sync(openRequestWaitQueue, ^ {
+        NSLog(@"Finished waitForOpenResponseLock");
```

Comment:
I will clean up the logs with some debug logging.

With the debug logging the block I think the code won't look weird.